services/src/datasets/upload.rs

```diff
@@ -39,11 +41,12 @@ impl Upload {
     }
 }
 
-#[derive(Debug, Serialize, Deserialize, Clone)]
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
+#[cfg_attr(feature = "postgres", derive(ToSql, FromSql))]
 pub struct FileUpload {
     pub id: FileId,
     pub name: String,
-    pub byte_size: usize,
+    pub byte_size: i64,
```

Comment:
Hm, is this a good strategy to change this type because of PostgreSQL?

Reply:
not the best bust most pragmatic. I could also implement the To/FromSQL manually or duplicate the type but with usize/isize, derive the traits and convert both into each other.

Reply:
I would say only an unsigned value makes sense, so either u64 or size.
You have more experience if implementing `To/FromSQL` is easier then using the DAO object.

Reply:
f76c432022b0dfab1387d83f20c5060c306187a2

I changed it to u64 but I couldn't get the >= 0 constraint to work on postgres side